blackduck/Client.py

```diff
@@ -201,28 +198,56 @@ def get_metadata(self, name, parent=None, **kwargs):
         return self.get_resource(name, parent, items=False, **kwargs)
 
     def checked_request(self, method, url, name='', **kwargs):
-        """Request from endpoint and return json result
+        """Send request to url endpoint and return json result
 
         Args:
             method (str): http request type e.g. 'get', 'post', etc.
             url (str): of endpoint
-            name (str, optional): for informational purposes in case of error. Defaults to ''.
+            name (str, optional): for informational purposes in case of error. Defaults to None.
             kwargs: passed to session.request
 
         Returns:
-            json/dict: requested object
-        """
-        response = self.session.request(method, url, **kwargs)
-
-        if response.status_code != 200:
-            self.http_exception_handler(response, name)
+            json/dict: requested object (status code 200) else None
 
```

Comment:
I removed the dependency on Exceptions.py for now as I needed logic based off the response.status_code to return a dict or None at the end.

Reply:
The current naming of 'checked_request' is very apt, but not at all what I intended when I initially wrote the function. Part of the reason I made `_request` private, was so that users wouldn't expect the public interface niceties. 

I would instead re-add `_request` as `raw_request` - with some changes to fully support post/put and still including the http_exception_handler, such that client dependents receive actionable errors rather than just messages.

Reply:
After further testing, POST and DELETE is best handled directly i.e. with bd.session.post and bd.session.delete. For example:

    project_data = {
        'name': "test-project",
        'description': "some description",
        'projectLevelAdjustments': True,
    }

    try:
        r = bd.session.post("/api/projects", json=project_data)
        r.raise_for_status()
        print(f"created project {r.links['project']['url']}")
    except requests.HTTPError as err:
        # more fine grained error handling here; otherwise:
        bd.log_http_error(err)

or this construct:

    r = bd.session.post("/api/projects", json=project_data)
    if r.status_code != 201:
        bd.log_http_error(r)
    else:
        print(f"created project {r.links['project']['url']}")

Having the user do some minimal form of error checking when POST'ing is to be expected. DELETE'ing can be done in a one line if the user simply ignores the result:

    bd.session.delete("/api/projects/fb23...")

As we have HubSession there is no longer a need to have a custom Client.post and Client.delete as it really does not add any necessary functionality nor does it simplify error handling.

That being said, when GET'ing data the following construct is used all the time:

    r = bd.session.get("/api/some_endpoint")
    r.raise_for_status()  # i.e. some kind of check _should_ be done...
    json_data = r.json()

Note this can be done in one line but error handling is very poor:

    json_data = bd.session.get("/api/some_endpoint").json()

Proper error handling dictates that we should log a useful error message if the http status_code != 200 or there is a JSONDecodeError. As this construct is ubiquitous, I renamed _request/checked_request to get_json and to handle only GET's. It allows the user to do a one-liner with nice error handling behind the scenes:

    json_data = bd.get_json("/api/some_endpont")

I think get_json() warrants being a publicly available method.